apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data.CommonTesting/SpannerTestDatabasePostgreSql.cs

```diff
@@ -0,0 +1,197 @@
+﻿// Copyright 2022 Google LLC
+// 
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// 
+//     https://www.apache.org/licenses/LICENSE-2.0
+// 
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using Google.Api.Gax;
+using Google.Api.Gax.ResourceNames;
+using Google.Cloud.ClientTesting;
+using Google.Cloud.Spanner.Admin.Database.V1;
+using Google.Cloud.Spanner.Admin.Instance.V1;
+using Google.Cloud.Spanner.Common.V1;
+using Google.Cloud.Spanner.V1.Internal.Logging;
+using Grpc.Core;
+using System;
+
+namespace Google.Cloud.Spanner.Data.CommonTesting
+{
+    /// <summary>
+    /// A database with PostgreSQL dialect created on-demand for testing. This is never dropped, partly as it's hard to know when to do so.
+    /// (This may be used by multiple fixtures, each created and then disposed, within the same test run.) 
+    /// </summary>
+    public class SpannerTestDatabasePostgreSql
+    {
+        private static readonly object s_lock = new object();
+        private static SpannerTestDatabasePostgreSql s_instance = null;
+
+        /// <summary>
+        /// Fetches the PostgreSQL database, creating it if necessary.
+        /// </summary>
+        /// <param name="projectId">The project ID to use, typically from a fixture.</param>
+        public static SpannerTestDatabasePostgreSql GetInstance(string projectId)
+        {
+            if (s_instance == null)
```

Comment:
This isn't performance sensitive - I'd suggest dropping double-checked locking for the sake of simplicity.
(Note that in its current form, the check for the project ID being incorrect would almost *never* fail, even if it should.)